typed-racket-lib/typed-racket/base-env/prims-contract.rkt

```diff
@@ -250,9 +251,17 @@
 ;;  make-predicate
 ;;  cast
 
-;; Helper to construct syntax for contract definitions
+;; Helpers to construct syntax for contract definitions
+;; make-contract-def-rhs : Type-Stx Boolean Boolean -> Syntax
 (define (make-contract-def-rhs type flat? maker?)
-  (contract-def-property #'#f `#s(contract-def ,type ,flat? ,maker? untyped)))
+  (define contract-def `#s(contract-def ,type ,flat? ,maker? untyped))
+  (contract-def-property #'#f (λ () contract-def)))
```

Comment:
I don't know, it might be impossible. I don't know very much about how type errors get reported here, how the contract generation phase works, or how the type-checker checks typed terms within `ignore-some` expressions, so I'm kind of paranoid. But still, lambda seems safer, even if promises are also safe in this case, the way the rest of the code is designed right now.